src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/ProxyConnectReplyStream.java

```diff
@@ -557,4 +564,19 @@ private int budgetAvailableWhenProxying(
             return 0;
         }
     }
+
+    private void checkEtag(EndFW end, CacheableRequest request)
+    {
+        final OctetsFW extension = end.extension();
+        if (extension.sizeof() != 0)
+        {
+            final HttpEndExFW httpEndEx = extension.get(streamFactory.httpEndExRO::wrap);
+            ListFW<HttpHeaderFW> trailers = httpEndEx.trailers();
+            String etag = trailers.matchFirst(h -> "etag".equals(h.name().asString())).value().asString();
+            assert etag !=null && !etag.isEmpty();
+            request.etag(etag);
+            request.setEtagInjected(true);
+            this.endExtension = extension;
```

Comment:
I am not sure if it is a good idea or not but if I receive etag as the trailer I am marking it as etagEnjected.

Reply:
What problem are you attempting to solve by doing this?
For example, what breaks if you don't do this?